### ISSUE TITLE
[Accessibility] StepList should include interactive example with links

### ISSUE THREAD

**Issue by majornista** [adobe/spectrum-css#215]
### Expected Behavior
StepList should provide an example that follows WAI tutorial recommendation for [Using a Step by Step indicator](https://www.w3.org/WAI/tutorials/forms/multi-page/#using-step-by-step-indicator).

### Actual Behavior
StepList only shows static examples. 
Does not have example where current step receives keyboard focus `:focus-within` or `.is-focused` on a descendant link.
Tooltip will not display on focus.

#### Browser name/version/os (ie Chrome Version 62.0.3202.94 (Official Build) (64-bit) MacOS)
Google Chrome Version 75.0.3770.100 (Official Build) (64-bit)

#### Spectrum-CSS version
v 2.13.0

#### Sample Code that illustrates the problem
```css
  /* Interactivity Styles */
  &.spectrum-Steplist--interactive {
    .spectrum-Steplist-item {
      /* Step Focused */
      &:focus {
        .spectrum-Steplist-marker {
          background-color: var(--spectrum-steplist-current-marker-color-key-focus)
        }
      }
    }
  }
```
Should be:
```css
  /* Interactivity Styles */
  &.spectrum-Steplist--interactive {
    .spectrum-Steplist-item {
      /* Step Focused */
      &:focus,
      &:focus-within,
      &.is-focused {
        .spectrum-Steplist-marker {
          background-color: var(--spectrum-steplist-current-marker-color-key-focus)
        }
      }
    }
  }
```
